packages/subproviders/src/subproviders/pk_wallet_subprovider.ts

```diff
@@ -0,0 +1,68 @@
+import { assert } from '@0xproject/assert';
+import { JSONRPCRequestPayload } from '@0xproject/types';
+import EthereumTx = require('ethereumjs-tx');
+import * as ethUtil from 'ethereumjs-util';
+import * as _ from 'lodash';
+
+import { Callback, ErrorCallback, PartialTxParams, ResponseWithTxParams, WalletSubproviderErrors } from '../types';
+
+import { BaseWalletSubprovider } from './base_wallet_subprovider';
+import { Subprovider } from './subprovider';
+
+/**
+ * This class implements the [web3-provider-engine](https://github.com/MetaMask/provider-engine) subprovider interface.
+ * This subprovider intercepts all account related RPC requests (e.g message/transaction signing, etc...)
+ */
+export class PKWalletSubprovider extends BaseWalletSubprovider {
+    private _address: string;
+    private _privateKeyBuffer: Buffer;
+    constructor(privateKey: string) {
+        super();
+        this._privateKeyBuffer = new Buffer(privateKey, 'hex');
+        this._address = `0x${ethUtil.privateToAddress(this._privateKeyBuffer).toString('hex')}`;
+    }
+    /**
+     * Retrieve the account calcuated from the private key.
+     * This method is automatically called when issuing a `eth_accounts` JSON RPC request
```

Comment:
`automatically` -> `implicitly`. I realize this is improving a comment I wrote earlier. Can't help but feel now that `automatically` is not as clear.